modules/fedimint-ln-tests/tests/tests.rs

```diff
@@ -124,7 +128,9 @@ async fn makes_internal_payments_within_federation() -> anyhow::Result<()> {
 #[tokio::test(flavor = "multi_thread")]
 async fn rejects_wrong_network_invoice() -> anyhow::Result<()> {
```

Comment:
Coincidentally I just started adding `#[instrument]` on every test and `[instrument(name = "<name>")]` on every binary `main` function.

We possibly could easily add a semgrep rule for that.

Reply:
This is good, but for it to be really useful we need to propagate the `span` to the server side (i.e add .instrument() on some futures)

Reply:
I guess I don't understand spans and how to use them effectively.

I thought that I can slap `#[instrument]` in major places and job done. :D

Reply:
In theory you should explicitly instrument every async function/code (https://docs.rs/tracing/latest/tracing/span/struct.Span.html#in-asynchronous-code).

But at least for most test purposes I found this is enough:
https://github.com/fedimint/fedimint/pull/3193/files#diff-476eca8e6f5bb7ac29c9bd2c6fbef5cff3c32b0f02ee2b7e24dffc7630eb7fb9R278

https://github.com/fedimint/fedimint/pull/3193/files#diff-6364ec89ecab2ba952666f453ab11412e5e0437503b88575358483403c20379bR120